### PR TITLE
Fix potential panic in patchStatefulsetReplicas

### PR DESCRIPTION
**Pull request by apapia**
Fix for https://github.com/govirtuo/kube-ns-suspender/issues/85

### CODE REVIEW

**Comment by apapia** [comment]
@eze-kiel let me know if I need to add a unit test or anything

File: engine/statefulset.go [hunk 1]
```diff
@@ -58,6 +58,9 @@ func patchStatefulsetReplicas(ctx context.Context, cs *kubernetes.Clientset, ns,
 		// if we want 0 replicas, it means that we are suspending the namespace,
 		// so before adjusting the replicas count, we want to save it for later
 		if repl == 0 {
+			if result.Annotations == nil {
```
Comment (eze-kiel) [inline engine/statefulset.go:61]: Please add a comment that explains what this block does, something like:
```go
if result.Annotations == nil {
	// there is no annotations in the object's manifest, so the map must
	// initialized
	// see issue #85
	result.Annotations = make(map[string]string)
}
```

**Review by eze-kiel — CHANGES_REQUESTED** [review]
Thanks for your contribution. Only 2 things are missing:

* a comment (see my review)
* the same edit on the `engine/deployment.go` file, that uses the same mechanism.

File: engine/statefulset.go (hunk 1 above)
Comment (apapia) [reply engine/statefulset.go:64]: good call

**Review by eze-kiel — APPROVED** [review]
LGTM, this closes #86

**Comment by eze-kiel** [comment]
Thanks @apapia 
The freshly built binary (and Docker image) should be up in a few minutes

**Pull request merged**